datafusion/core/tests/parquet/mod.rs

```diff
@@ -184,7 +184,13 @@ impl TestOutput {
 /// and the appropriate scenario
 impl ContextWithParquet {
     async fn new(scenario: Scenario, unit: Unit) -> Self {
-        Self::with_config(scenario, unit, SessionConfig::new()).await
+        let mut session_config = SessionConfig::new();
+        // TODO (https://github.com/apache/datafusion/issues/12817) once this is the default behavior, remove from here
```

Comment:
Does this means that DataFusion will no longer prune predicates like `decimal_col = 5.0`?

If so, this like a significant regression / issue for anyone who relies on decimal types (like comet for example)

Reply:
I am not sure if this will affect pruning - but it'd be great if @andygrove can take a look

Reply:
I will look at this today.